test_util/marathon.py

```diff
@@ -205,6 +205,83 @@ def _pool_for_marathon_app(app_id):
             raise Exception("Application deployment failed - operation was not "
                             "completed in {} seconds.".format(timeout))
 
+    def deploy_pod(self, pod_definition):
+        """Deploy a pod to marathon
+
+        This function deploys an a pod and then waits for marathon to
+        aknowledge it's successfull creation or fails the test.
+
+        It waits until all the instances reach tasksRunning and then tasksHealthy state.
+
+        Args:
+            pod_definition: a dict with pod definition as specified in
+                            Marathon API
+            check_health: wait until Marathon reports tasks as healthy before
+                          returning
+        Returns:
+            Scaling instance count
+        """
+        timeout = 120
+
+        r = self.post('v2/pods', json=pod_definition)
+        logging.info('Response from marathon: {}'.format(repr(r.json())))
```

Comment:
nit: the response body might not be json which would cause this to error badly (say there was a permission denied because the JWT token expired so the API response comes from bouncer and isn't json). Putting this after the r.ok would probably make this easier to debug.